turbojpeg.py

```diff
@@ -260,7 +291,63 @@ def scale_with_quality(self, jpeg_buf, scaling_factor=None, quality=85, flags=0)
         finally:
             self.__destroy(handle)
 
+    def crop(self, jpeg_buf, x, y, w, h, gray=False):
+        """losslessly crop a jpeg image with optional grayscale"""
+        handle = self.__init_transform()
+        try:
+            jpeg_array = np.frombuffer(jpeg_buf, dtype=np.uint8)
+            src_addr = self.__getaddr(jpeg_array)
+            width = c_int()
+            height = c_int()
+            jpeg_colorspace = c_int()
+            jpeg_subsample = c_int()
+            status = self.__decompress_header(
+                handle, src_addr, jpeg_array.size, byref(width), byref(height),
+                byref(jpeg_subsample), byref(jpeg_colorspace))
+            if status != 0:
+                self.__report_error(handle)
+            if x < 0:
+                x = 0
+            else:
+                x = (x / 16) * 16
+            if y < 0:
+                y = 0
+            else:
+                y = (y / 16) * 16
+            if w < 0:
+                w = 0
```

Comment:
"Setting this to 0 is the equivalent of setting it to the height of the source JPEG image - y"  from https://cdn.rawgit.com/libjpeg-turbo/libjpeg-turbo/master/doc/html/structtjregion.html#aecefc45a26f4d8b60dd4d825c1710115

I don't know that you need to change anything here...but that is a bit surprising.